core/src/test/java/jenkins/util/UrlHelperTest.java

```diff
@@ -10,56 +10,140 @@
+        assertTrue(UrlHelper.isValidRootUrl("http://172.52.125.12:8080/jenkins"));
+    }
+    
+    @Test
+    public void ipv6Allowed() {
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]"));
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:0000:0000:3210:FEDC:BA98:7654:3210]"));
+        // 0000 can be reduced to 0
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:0:0:3210:FEDC:BA98:7654:3210]"));
+        // an unique sequence of multiple fragments with 0's could be omitted completely
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC::3210:FEDC:BA98:7654:3210]"));
+        // but only one sequence
+        assertFalse(UrlHelper.isValidRootUrl("http://[2001::85a3::ac1f]"));
+        
+        // port and path are still allowed
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:0:0:3210:FEDC:BA98:7654:3210]:8001/jenkins"));
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:0:0:3210:FEDC:BA98:7654:3210]:8001"));
+        assertTrue(UrlHelper.isValidRootUrl("http://[FEDC:0:0:3210:FEDC:BA98:7654:3210]/jenkins"));
+        // dashes are not allowed inside ipv6
+        assertFalse(UrlHelper.isValidRootUrl("http://[FEDC:0:0:32-10:FEDC:BA98:7654:3210]:8001/jenkins"));
+        assertFalse(UrlHelper.isValidRootUrl("http://[FEDC:0:0:3210:-FEDC:BA98:7654:3210]:8001/jenkins"));
     }
     
     @Test
     @Issue("JENKINS-51064")
-    public void isAlsoValid() {
+    public void withCustomDomain() {
         assertTrue(UrlHelper.isValidRootUrl("http://my-server:8080/jenkins"));
         assertTrue(UrlHelper.isValidRootUrl("http://jenkins.internal/"));
         assertTrue(UrlHelper.isValidRootUrl("http://jenkins.otherDomain/"));
         assertTrue(UrlHelper.isValidRootUrl("http://my-server.domain:8080/jenkins"));
         assertTrue(UrlHelper.isValidRootUrl("http://my-ser_ver.do_m-ain:8080/jenkins"));
         assertTrue(UrlHelper.isValidRootUrl("http://my-ser_ver.do_m-ain:8080/jenkins"));
-    
+        
         // forbidden to start or end domain with - or .
         assertFalse(UrlHelper.isValidRootUrl("http://-jenkins.com"));
         assertFalse(UrlHelper.isValidRootUrl("http://jenkins.com-"));
         assertFalse(UrlHelper.isValidRootUrl("http://.jenkins.com"));
-        assertFalse(UrlHelper.isValidRootUrl("http://jenkins.com."));
+        
+        // allowed to have multiple dots in chain
+        assertTrue(UrlHelper.isValidRootUrl("http://jen..kins.com"));
+    }
     
-        // forbidden to have multiple dots in chain
-        assertFalse(UrlHelper.isValidRootUrl("http://jen..kins.com"));
+    @Test
+    public void multipleConsecutiveDashesAreAllowed() {
+        assertTrue(UrlHelper.isValidRootUrl("http://jenk--ins.internal/"));
+        assertTrue(UrlHelper.isValidRootUrl("http://www.go-----ogle.com/"));
+        // even with subdomain being just a dash
+        assertTrue(UrlHelper.isValidRootUrl("http://www.go.-.--.--ogle.com/"));
```

Comment:
This gives me ideas on subdomains to start using. 😈

Reply:
Just the leading ..... was a bit weird initially ^^ but when you learn that simply means you explicitely write down the Root domain which is "", that's seems clearer

FTR I am not sure the single dash sub-domains are valid outside of some weird / custom / internal DNS configuration

Reply:
You should see all the strange email addresses that are technically valid! Reminds me of the same scenario where some technically valid hosts won't be recognized by certain programs.